llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2062,6 +2062,26 @@ static Instruction *foldSubOfMinMax(BinaryOperator &I,
   return nullptr;
 }
 
+/// Fold `sub 0, (udiv nneg X, nneg C)` into `sdiv nneg X, -C`
```

Comment:
You can handle some additional cases if we have `exact` flag:
https://alive2.llvm.org/ce/z/mmpBoS

Reply:
If I'm not wrong, the second example should make sense if RHS is not a constant.

Reply:
I don't think so, in that case you aren't saving any instructions and basically just throwing out `exact` and doing `udiv`-> `sdiv`. Both of which alone are not profitable.